### ISSUE TITLE
GLPI is sending uselessly too many queries to database.

### ISSUE THREAD

**Issue by ettalhi** [glpi-project/glpi#5273]
**Describe the bug**
GLPI is sending uselessly too many queries to database; the same query could be repeated too many times during the same HTTP request. The number of the queries can reach easily tens of thousands of times which may affect greatly GLPI performance.

**To reproduce**
1. Create new object via [GLPI/Setup/Objects management].
2. Let's say we have created the following new objects :
![new_objects_created](https://user-images.githubusercontent.com/29466957/51398226-0ee82680-1b11-11e9-95d9-2c244459ecb8.jpg)
3. Let's go to [GLPI/Setup/Fields unicity] with debug mode enabled. I have already threes rules there linked to Bicyclette object. I see now [649 Queries took 0.011s] when I click on debugger button. see queries in file chevre_champ_unicite_00.txt attached below.
[chevre_champ_unicite_00.txt](https://github.com/glpi-project/glpi/files/2774177/chevre_champ_unicite_00.txt)
In the log file [chevre_champ_unicite_00.txt], we can notice that the following two SQL queries have been repeated 261 times each :
SELECT `id`
FROM `glpi_plugins`
WHERE `glpi_plugins`.`directory` = 'datainjection'
SELECT *
FROM `glpi_plugins`
WHERE `glpi_plugins`.`id` = '5' LIMIT 1
4. Even though we have already seen the problem in point (3), we will see how adding a new [Fields unicity] rule linked to object [Chevre] will increase the number of queries sent to DB. So Let's add a new rule [Regle_05] linked to [Chevre] object under the same location [GLPI/Setup/Fields unicity].
![screen_toomany_queries](https://user-images.githubusercontent.com/29466957/51410527-e1ac7000-1b32-11e9-82ff-7c38dd7e717e.jpg)
5. Return to the page  [GLPI/Setup/Fields unicity] after creating the new rule and see how many SQL queries do we have in debugging mode. (see file chevre_champ_unicite_01.txt attached)
[chevre_champ_unicite_01.txt](https://github.com/glpi-project/glpi/files/2774179/chevre_champ_unicite_01.txt)
The log has increased from 649 before to 805 queries by repeating uselessly just the two queries mentioned above.
We can see this time that the following two SQL queries have been repeated 339 times each :
SELECT `id`
FROM `glpi_plugins`
WHERE `glpi_plugins`.`directory` = 'datainjection'
SELECT *
FROM `glpi_plugins`
WHERE `glpi_plugins`.`id` = '5' LIMIT 1

We can also see the following query repeated unnecessarily many times:
SELECT `TABLE_NAME`
FROM `information_schema`.`TABLES`
WHERE `TABLE_SCHEMA` = 'glpidevn1' AND `TABLE_TYPE` = 'BASE TABLE' AND `TABLE_NAME` LIKE '%criteria%'
for different values of [criteria].
6. We can add another unicity rule and see again the increase in the number of the queries. sometimes the repeated queries are different from those above depending on which screen of GLPI we are.

**Expected behavior**
We want those queries above to run only once, or at least a reasonable number of times. By this we will avoid an exponentially increase in the number of SQL queries sent to DB which would slow down considerably the system.

**Suggested solution**
1. In the file GLPI_ROOT\inc\plugin.class.php, I have modified the two functions function isActivated($plugin) and function isInstalled($plugin). And in the file opt\bitnami\apps\glpi\htdocs\inc\dbmysql.class.php, the function public function tableExists_new($tablename) , as shown in the attached solution_01.txt :
[solution_01.txt](https://github.com/glpi-project/glpi/files/2774308/solution_01.txt)
2. I refresh the same form [GLPI/Setup/fields unicity] (page : http://10.150.194.74/glpi/front/fieldunicity.php ) to see the gain; the queries sent to DB have passed from 805 to 96 ! the log after correction is attached below.
[chevre_champ_unicite_02.txt](https://github.com/glpi-project/glpi/files/2774333/chevre_champ_unicite_02.txt)
and these are some remarks :
- The query 
[SELECT `id`
FROM `glpi_plugins`
WHERE `glpi_plugins`.`directory` = 'datainjection'] is repeated only twice instead of 339 before correction.
- The query fragment
[SELECT *
FROM `glpi_plugins`
WHERE `glpi_plugins`.`id` = ] is repeated only 8 times instead of 346.
- The query [
SELECT `TABLE_NAME`
FROM `information_schema`.`TABLES`
WHERE `TABLE_SCHEMA` = 'glpidevn1' AND `TABLE_TYPE` = 'BASE TABLE' AND `TABLE_NAME` LIKE '%glpi_plugin_genericobject_types%'] is repeated only 1 time instead of 4 before.

**Comment by trasher** [comment]
Hi,

Please open a pull request against master with your proposed changes; unit tests will be run, and we can inspect your proposition. Thank you :)

**Comment by trasher** [comment]
OK, this kid of way sounds strange to us, but this is not a problem.

As I've said on the PR, CI must be enabled, or PR should be opened directly on GLPI repo so CI can run.

**Comment by loicbenoit** [comment]
Tests are failing because the test apparatus can't connect to the test DB. The errors have nothing to do with the submitted code.

**Comment by trasher** [comment]
Tests are OK without your changes. You have an SQL error in it.

**Comment by cedric-anne** [comment]
Hi,

I made some improvements on 9.4/bugfixes branch for plugins state checks, especially on #5545 which can limit queries to the glpi_plugins table.

Could you check if that fixes this problem ?

Regards

**Comment by cedric-anne** [comment]
Hi,

I consider that this performance issue has been adressed in last iteration we made on cache usage and plugins management.
So I close this issue.

Regards

**Comment by ettalhi** [comment]
Hello Cedric,
I hope you will hold on with the merge of this enhancement, because I have just discovered an issue with it.
I will fix it in a separate PR and bind it to this same issue .
The issue occurs in installation mode.
Thanks.

**Comment by ettalhi** [comment]
[The last PR (17) attached to this issue ](https://github.com/VilledeMontreal/glpi/pull/17) solves only a part of the reported issue since the other part has been solved by community in subsequent version 9.4
